### PR TITLE
Update/c++ 

### PR DESCRIPTION
**Pull request by bayars**
i have added c++ code for hacker rank practice. Can you merge it now?

### CODE REVIEW

File: HackerRank/Python/Data Types/Find second largest number in a list.py [hunk 1]
```diff
@@ -1,25 +1,13 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-'''
-Problem Statement
+if __name__ == '__main__':
```
Comment (ad1992) [inline HackerRank/Python/Data Types/Find second largest number in a list.py]: can you keep the comments as it was previously ? And keep the prev solution as well and add this as an alternate solution in a a new file

File: HackerRank/c++/ArraysIntroduction.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    cout << a[i] << " ";
+}
+//     for(int m=0;m<limit;m++){
+//     for(int t=limit;0<=t;--t){
+//             l[m]=a[t];
+//     }  }
+  
+//     // for(int i=0;i<limit;i++){
+//     //     cout << l[i];
+//     // }
```
Comment (ad1992) [inline HackerRank/c++/ArraysIntroduction.cpp]: remove the commented code

**Review by ad1992 — CHANGES_REQUESTED** [review]
@rection thanks for your contribution. can you add a link to the challenge in every solution ? you can add a readme for each challenge with the link and sample  input/output. Also can you give the submission link for each solution in the PR  so that I can view the submission status of the solution before merging

File: HackerRank/Python/Data Types/Find second largest number in a list.py (hunk 1 above)
Comment (ad1992) [reply HackerRank/Python/Data Types/Find second largest number in a list.py]: Also can you remove the binary `.out` files from the PR and add in gitignore as well

**Comment by bayars** [comment]
- [Say "Hello, World!" With C++](https://www.hackerrank.com/challenges/cpp-hello-world/submissions/code/86795724)
- [Input And Output](https://www.hackerrank.com/challenges/cpp-input-and-output/submissions/code/86843279)
- [Basic Data Types](https://www.hackerrank.com/challenges/c-tutorial-basic-data-types/submissions/code/87031954)
- [Conditional Statements](https://www.hackerrank.com/challenges/c-tutorial-conditional-if-else/submissions/code/87032273)
- [For Loop](https://www.hackerrank.com/challenges/c-tutorial-for-loop/submissions/code/87041323)
- [Functions](https://www.hackerrank.com/challenges/c-tutorial-functions/submissions/code/87203395)
- [Pointer](https://www.hackerrank.com/challenges/c-tutorial-pointer/submissions/code/87204786)
- [ArraysIntroduction](https://www.hackerrank.com/challenges/arrays-introduction/submissions/code/87206862)
- [String Stream](https://www.hackerrank.com/challenges/c-tutorial-stringstream/submissions/code/87428275)
- [Strings](https://www.hackerrank.com/challenges/c-tutorial-strings/submissions/code/87429125)
- [Structs](https://www.hackerrank.com/challenges/c-tutorial-struct/submissions/code/87429687)
- [Class](https://www.hackerrank.com/challenges/c-tutorial-class/submissions/code/87432451)
- [Classes and Objects](https://www.hackerrank.com/challenges/classes-objects/submissions/code/87512211)
- [Box it!](https://www.hackerrank.com/challenges/box-it/submissions/code/87514579)


ı did what you say. Now you can merge

File: HackerRank/Python/Data Types/Find second largest number in a list.py (hunk 1 above)
Comment (bayars) [reply HackerRank/Python/Data Types/Find second largest number in a list.py]: @ad1992  did it.

File: HackerRank/c++/ArraysIntroduction.cpp (hunk 2 above)
Comment (bayars) [reply HackerRank/c++/ArraysIntroduction.cpp]: @ad1992 i did it.

File: HackerRank/Python/Data Types/Find second largest number in a list.py [hunk 3]
```diff
@@ -1,25 +1,14 @@
...
 NOTE : Don't use insertion sort
-
 Input Format
-
 First line contains N. Second line contains Array A[] of N integers each separated by a space.
-
 Output Format
-
 Value of the second largest number.
 '''
```
Comment (ad1992) [inline HackerRank/Python/Data Types/Find second largest number in a list.py:11]: typo the comment has no starting point so will lead to error when this code is executed. Also as you created the read me with the links for other challenges , can you do the same thing for this challenge as well so the problem statement can be removed from the code

File: HackerRank/Python/Data Types/Find second largest number in a list.py [hunk 4]
```diff
@@ -1,25 +1,14 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
```
Comment (ad1992) [inline HackerRank/Python/Data Types/Find second largest number in a list.py]: revert

File: HackerRank/Python/Data Types/Find second largest number in a list.py [hunk 5]
```diff
@@ -1,25 +1,14 @@
-#!/usr/bin/env python
```
Comment (ad1992) [inline HackerRank/Python/Data Types/Find second largest number in a list.py]: revert

File: HackerRank/Python/Data Types/Find second largest number in a list.py [hunk 6]
```diff
@@ -1,25 +1,14 @@
...
 '''
 
 N=int(raw_input())
 list=list(set(map(int,raw_input().strip().split(" "))))
 list.sort(reverse=True)
 print list[1]
-
-    
-    
+    print(arr[0])
```
Comment (ad1992) [inline HackerRank/Python/Data Types/Find second largest number in a list.py]: remove

**Review by ad1992 — CHANGES_REQUESTED** [review]
looks good to merge. can you fix the python code and remove .gecici.swp as well and rebase with master so that conflicts are resolved.

File: HackerRank/Python/Data Types/Find second largest number in a list.py (hunk 6 above)
Comment (bayars) [reply HackerRank/Python/Data Types/Find second largest number in a list.py]: @ad1992 i have added your code in "Find second largest number in a list.py" and  "Find second largest number in a list-2.py"  file is mine. I was removed .gecici.swp.

**Comment by ad1992** [comment]
@rection the branch is not yet rebased with master branch hence shows the conflicts. Can you pull the latest master and merge your branch with latest master

**Comment by bayars** [comment]
@ad1992 i have merged branch.

**Comment by ad1992** [comment]
@rection its not yet rebased with master, please see the below screenshot
screen shot 2018-10-14 at 7 48 37 pm
You need to fetch the upstream branch and merge your branch and push

**Comment by bayars** [comment]
@ad1992  i have applied rebase for master can you check again.

**Comment by ad1992** [comment]
@rection you haven't pulled the upstream hence it shows the same issue
screen shot 2018-10-15 at 11 43 17 am

the above screenshot should only say ahead of 10 commits.
@rection try executing the below commands
```
git checkout master
git fetch upstream
git merge upstream/master
git push origin master
```
This should fix your branch unless there are conflicts and then you can rebase with this branch with your master branch

**Comment by ad1992** [comment]
@rection doesn't seem like you pushed master to your branch. can you check the commands and fix your master branch so that it is only ahead and not behind any commit and then rebase this branch. your pr still shows conflicts

**Comment by bayars** [comment]
gonder

@ad1992  I have make your say but it didn't. I think, PR opened on c++/update branch so you may merge on only c++/update. after that you merge c++/update branch to master branch. 

This picture for master branch:
ad1992  
This picture for c++/update branch:
  

also i try solve 'This branch is 11 commits ahead, 8 commits behind ad1992:master. ' but i saw a output. I think, commits send  HEAD branch. HEAD branch purpose is save commit. I tried reset head branch but it didn't work. Head branch work [reason](https://git-scm.com/book/en/v1/Git-Branching-What-a-Branch-Is).



'''
How does Git know what branch you’re currently on? It keeps a special pointer called HEAD. Note that this is a lot different than the concept of HEAD in other VCSs you may be used to, such as Subversion or CVS. In Git, this is a pointer to the local branch you’re currently on. In this case, you’re still on master. The git branch command only created a new branch — it didn’t switch to that branch (see Figure 3-5).
'''

**Comment by ad1992** [comment]
@rection thanks for your contribution. All good now and merged :)

**Pull request merged**